cpp/src/arrow/compute/CMakeLists.txt

```diff
@@ -142,7 +182,9 @@ add_arrow_compute_test(row_test
                        EXTRA_LINK_LIBS
                        arrow_compute_testing)
 
-add_arrow_benchmark(function_benchmark PREFIX "arrow-compute")
+if(ARROW_BUILD_BENCHMARKS)
```

Comment:
Can we remove this `if(ARROW_BUILD_BENCHMARKS)`?
We want to do it only in `add_benchmark()`.

Reply:
Thanks for sharing existing examples.

The last example case is redundant. We can simply remove `if(ARROW_BUILD_BENCHMARK)`.

https://github.com/apache/arrow/blob/953947a78157090ff94b14bc027cbcf520a5b2a6/cpp/src/arrow/filesystem/CMakeLists.txt#L35-L42

Other example cases: I think that we can remove `if(ARROW_BUILD_BENCHMARKS)` by adding `EXTRA_PUBLIC_LINK_LIBS` to `add_arrow_benchmark()`.

I can do it as a separated task later. So you don't need to care about it in this PR.

Reply:
ok, I understand now, I didn't know that the `NO_BENCHMARKS` variable was basically mirroring `ARROW_BUILD_BENCHMARKS` and acting as the guard to build those on `add_benchmark`, all clear now, I've removed the unnecessary if checks.